packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -594,21 +599,42 @@ export function inspectHooks<Props>(
   return buildTree(rootStack, readHookLog);
 }
 
-function setupContexts(contextMap: Map<ReactContext<any>, any>, fiber: Fiber) {
+function setupContexts(
```

Comment:
The overloading of this function (sometimes a return type, sometimes not) and the newly-added `enableLegacyContext` flag, seems a little confusing to me. Let's split them up?

```js
function getLegacyContext(fiber: Fiber): Object | null {
  const hasLegacyContext = !!fiber.elementType.contextTypes;
  if (hasLegacyContext) {
    let current = fiber;
    while (current !== null) {
      const instance = current.stateNode;
      if (instance && instance.__reactInternalMemoizedMergedChildContext) {
        return instance.__reactInternalMemoizedMergedChildContext;
      }
      current = current.return;
    }
  }
  return null;
}

function setupContext(contextMap: Map<ReactContext<any>, any>, fiber: Fiber) {
  let current = fiber;
  while (current) {
    if (current.tag === ContextProvider) {
      const providerType: ReactProviderType<any> = current.type;
      const context: ReactContext<any> = providerType._context;
      if (!contextMap.has(context)) {
        // Store the current value that we're going to restore later.
        contextMap.set(context, context._currentValue);
        // Set the inner most provider value on the context.
        context._currentValue = current.memoizedProps.value;
      }
    }
    current = current.return;
  }
}
```